libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -358,15 +404,17 @@ class MapboxRouteLineApi(
      * @return a state that represents the side effects that when rendered will
      * set the alternative route(s) to a state of not visible.
      */
-    fun hideAlternativeRoutes(): RouteLineState.UpdateLayerVisibilityState {
-        return RouteLineState.UpdateLayerVisibilityState(
-            listOf(
-                Pair(RouteLayerConstants.ALTERNATIVE_ROUTE1_LAYER_ID, Visibility.NONE),
-                Pair(RouteLayerConstants.ALTERNATIVE_ROUTE1_CASING_LAYER_ID, Visibility.NONE),
-                Pair(RouteLayerConstants.ALTERNATIVE_ROUTE2_LAYER_ID, Visibility.NONE),
-                Pair(RouteLayerConstants.ALTERNATIVE_ROUTE2_CASING_LAYER_ID, Visibility.NONE),
-                Pair(RouteLayerConstants.ALTERNATIVE_ROUTE1_TRAFFIC_LAYER_ID, Visibility.NONE),
-                Pair(RouteLayerConstants.ALTERNATIVE_ROUTE2_TRAFFIC_LAYER_ID, Visibility.NONE)
+    fun hideAlternativeRoutes(): Expected<LayerVisibilityUpdate, RouteLineError> {
```

Comment:
Shouldn't this be executed directly on the `RouetLineView`?

We've been discussing managing visibility on the views themselves. Same for `show`.

Reply:
updated